impl/src/main/java/org/ehcache/config/builders/CacheManagerBuilder.java

```diff
@@ -212,4 +230,52 @@ T cast(EhcacheManager ehcacheManager) {
   public static PersistenceConfiguration persistence(String location) {
     return new CacheManagerPersistenceConfiguration(new File(location));
   }
+
+  /**
+   * Creates a {@code CacheManager} of the designated type from the configuration provided.
+   * Configurations passed to this method must have either no more than one
+   * {@link CacheManagerConfiguration} present in the
+   * {@linkplain Configuration#getServiceCreationConfigurations() service creation configuration list}
+   * or the {@code CacheManagerConfiguration} instances must be combinable.  When neither of these
+   * constraints is met, the result of this method is not defined.
+   *
+   * @param specialization the type of the desired {@code CacheManager}
+   * @param config the {@code Configuration} to use in the {@code CacheManager}
+   * @param <T> the type of the desired {@code CacheManager}
+   *
+   * @return a new {@code CacheManager} instance of the type designated by {@code specialization};
+   *      the returned {@code CacheManager} is <b>not</b> initialized
+   *
+   * @throws ClassCastException if the {@link CacheManagerBuilder} associated with the
+   *        {@link CacheManagerConfiguration} instances appearing in the configuration does
+   *        not build a {@code CacheManager} of the {@code specialization} type
+   */
+  public static <T extends CacheManager> T newSpecializedCacheManager(Class<T> specialization, Configuration config)
```

Comment:
It's presently used from the `org.ehcache.clustered.ClusteredCacheManagerBuilder` class ...